src/petals/server/server.py

```diff
@@ -469,7 +495,8 @@ def __init__(
             )
             for _ in range(num_handlers)
         ]
-        self.runtime = Runtime(self.module_backends, **kwargs)
+        self.runtime = Runtime(self.module_backends, device=None, **kwargs)
+        # note: we set device=None in runtime to avoid casting all modules to device 0 in .run; we'll cast them manually
```

Comment:
```suggestion
        # note: We set device=None in runtime to avoid casting all modules to device 0 in runtime.run(). tensor_parallel will cast it for us as needed.
```